src/iterators/quantile.rs

```diff
@@ -21,33 +21,84 @@ impl<'a, T: 'a + Counter> Iter<'a, T> {
                                    hist,
                                    ticks_per_half_distance,
                                    quantile_to_iterate_to: 0.0,
-                                   quantile_just_picked: 0.0
+                                   reached_end: false
                                })
     }
 }
 
 impl<'a, T: 'a + Counter> PickyIterator<T> for Iter<'a, T> {
-    fn pick(&mut self, index: usize, running_total: u64) -> bool {
-        let count = &self.hist.count_at_index(index)
-            .expect("index must be valid by PickyIterator contract");
-        if *count == T::zero() {
-            return false;
+    fn pick(&mut self, _: usize, running_total: u64, count_at_index: T) -> Option<PickMetadata> {
+        if count_at_index == T::zero() {
+            return None;
         }
 
         // This calculation, combined with the `quantile * count` in `value_at_quantile`, tends
         // to produce a count_at_quantile that is 1 ulp wrong. That's just the way IEEE754 works.
         let current_quantile = running_total as f64 / self.hist.count() as f64;
         if current_quantile < self.quantile_to_iterate_to {
-            return false;
+            return None;
+        }
+
+        // Because there are effectively two quantiles in play (the quantile of the value for the
```

Comment:
comments per LoC might be getting a little out of hand in this struct, but I like to beat historically confusing logic into submission with overwhelming documentation

Reply:
I have no objection to this comment. Seems well-placed and useful.